rust/ethers-prometheus/src/middleware/mod.rs

```diff
@@ -0,0 +1,564 @@
+//! A middleware layer which collects metrics about operations made with a
```

Comment:
I didn't review this file - just wanna make sure this is just copied from `rust/ethers-prometheus/src/lib.rs`

Reply:
Mostly, there were some minor changes but you saw the result of those elsewhere.